### PR TITLE
expose httpclient as public

### PR DESCRIPTION
**Pull request by mjlamb**
this allow use of custom httpclient with handler

### CODE REVIEW

File: NBitpayClient/BitPay.cs [hunk 1]
```diff
@@ -157,16 +156,25 @@ public void SaveToken(string key, string value)
...
 
-
-
-
 		private const String BITPAY_API_VERSION = "2.0.0";
 
 		private Uri _baseUrl = null;
 		AuthInformation _Auth;
-		static HttpClient _Client = new HttpClient();
+        public static HttpClient _httpClient = new HttpClient();
```
Comment (NicolasDorier) [inline NBitpayClient/BitPay.cs]: the shared instance must be private.

File: NBitpayClient/BitPay.cs [hunk 2]
```diff
@@ -157,16 +156,25 @@ public void SaveToken(string key, string value)
...
+
+        public HttpClient HttpClient
+        {
+            get
+            {
+                return _httpClient;
+            }
+            set
+            {
+                _httpClient = HttpClient;
```
Comment (NicolasDorier) [inline NBitpayClient/BitPay.cs]: this is equivalent to `_httpClient = _httpClient` which is doing nothing

File: NBitpayClient/BitPay.cs [hunk 3]
```diff
@@ -671,7 +679,7 @@ private async Task<HttpResponseMessage> GetAsync(String path, bool sign)
 				{
 					_Auth.Sign(message);
 				}
-				var result = await _Client.SendAsync(message).ConfigureAwait(false);
+				var result = await _httpClient.SendAsync(message).ConfigureAwait(false);
```
Comment (NicolasDorier) [inline NBitpayClient/BitPay.cs]: this always use the shared instance

File: NBitpayClient/BitPay.cs (hunk 1 above)
Comment (mjlamb) [reply NBitpayClient/BitPay.cs]: made private

File: NBitpayClient/BitPay.cs (hunk 2 above)
Comment (mjlamb) [reply NBitpayClient/BitPay.cs]: corrected to use value being passed in

File: NBitpayClient/BitPay.cs (hunk 3 above)
Comment (mjlamb) [reply NBitpayClient/BitPay.cs]: Do not know what you mean. I'm using the private set property here either that set by external public property or the internal new up of HttpClient against private property.

File: NBitpayClient/BitPay.cs [hunk 4]
```diff
@@ -170,7 +170,7 @@ public HttpClient HttpClient
             }
             set
             {
-                _httpClient = HttpClient;
```
Comment (NicolasDorier) [inline NBitpayClient/BitPay.cs]: You are now modifying a `static` value from an instance property. 
This is not what you want. You want that by default the instance value is the shared HttpClient, but still allow somebody else to override the **instance** value to something else.

**Pull request closed without merging**